### DISCUSSION TITLE
0.3

### DISCUSSION THREAD

**Discussion by EmiOnGit** [EmiOnGit/warbler_grass#47] [Announcements]
# Changelog

### Change
* The `GrassSpawner` was removed in favor of the `WarblersBundle` and `WarblersExplicitBundle`
* The `wind_noise_texture` field of the `GrassConfiguration` resource was moved to the `GrassNoiseTexture` resource
* The `density_map` and `height_map` modules were combined to the `maps` module
* Dithering uses a new matrix which changes the positions of your grass blades slightly if you used a density map before

### Added
* A editor was implemented
This makes it possible to modify your maps directly in the game and save them after
The editor can only be used with the new `editor` feature (you can take a look at the editor example)
* A `WarblerDiagnosticsPlugin` has been added which logs the amount of grass bladed rendered every frame
This is used in the `many_chunks` exampe if you'd like to see how to use it
* The height of the blades can now be sampled using a texture with the `WarblersHeight::Texture` component
* A density parameter can now also be specified with the `DensityMap` to directly influence the dithering
* The default wind direction was changed from (1,0) to (1,1)

### Fix
Many bugs have been fixed; to many to count all of them
* Warbler_grass only uses the dependencies it needs!
* Grass chunks are updated now also updated if the height of the blades changed
* The frustum culling of chunks now works properly with big chunks
* A grass chunk is now set invisible if the parent is also invisible
* We don't crash the program if the density map is removed
* If hot reloading is activated, all maps are monitored
* Wind doesn't get faster over time

<hr /><em>This discussion was created from the release <a href='https://github.com/EmiOnGit/warbler_grass/releases/tag/v0.3'>0.3</a>.</em>

**Comment by janhohenheim** [comment]
I've updated to 0.3 and changed my spawning code to use the `WarblersExplicitBundle`, since that seemed like the closest thing to my existing grass generation code. However, now my grass doesn't show up anymore 😱 
Code: https://github.com/janhohenheim/foxtrot/pull/228/files#diff-f7cf21045cf163dc0a11d0ebf9b1d7541c14da67855b984f66147b482cbfeed9L48-L66
![image](https://user-images.githubusercontent.com/9047632/225042501-9341f92f-a381-482b-a1db-e59ca12e03fd.png)
Looking at the generated positions, they seem valid to me:
![image](https://user-images.githubusercontent.com/9047632/225042709-5a6040c5-3258-48d4-8565-5542a6013d6d.png)

Any idea on what I might be doing wrong? Does this have something to do with the grass not being visible from the start and thus somehow not entering a cache?

**Reply by EmiOnGit** [reply]
Interesting, I'll look into that.
What do mean by not visible from the start?
